sdk/monitor/monitor-ingestion/test/public/logsIngestionClient.spec.ts

```diff
@@ -97,47 +95,133 @@ describe("LogsIngestionClient live tests", function () {
-      assert.isAbove(chunkArraySize, 1);
-      if (chunkArraySize % 2 === 0) {
-        assert.equal(result.errors.length, chunkArraySize / 2);
-      }
-      if (chunkArraySize % 2 === 1) {
-        assert.equal(result.errors.length, (chunkArraySize - 1) / 2);
+        const chunkArraySize = getChunkArraylength(noOfElements);
+        assert.isAbove(chunkArraySize, 1);
+        if (chunkArraySize % 2 === 0) {
+          assert.equal(result.length, chunkArraySize / 2);
+        }
+        if (chunkArraySize % 2 === 1) {
+          assert.equal(result.length, (chunkArraySize - 1) / 2);
+        }
       }
     }
   });
 
   it("Throws error when all logs fail", async function () {
     const noOfElements = 25000;
     const logData = getObjects(noOfElements);
-    const result = await client.upload("immutable-id-123", "Custom-MyTableRawData", logData, {
-      maxConcurrency: 3,
-    });
-    assert.equal(result.status, "Failure");
-    if (result.status !== "Success") {
-      result.errors.forEach((err) => {
+    try {
+      await client.upload("immutable-id-123", "Custom-MyTableRawData", logData, {
+        maxConcurrency: 3,
+      });
+    } catch (e: any) {
+      const result = isAggregateUploadLogsError(e) ? e.errors : [];
+      if (result.length > 0) {
+        result.forEach((err) => {
+          assert.equal(
+            err.cause.message,
+            `Data collection rule with immutable Id 'immutable-id-123' not found.`
+          );
+        });
+        const chunkArraySize = getChunkArraylength(noOfElements);
+        assert.isAbove(chunkArraySize, 1);
+        assert.equal(chunkArraySize, result.length);
+      }
+    }
+  });
+  it("Calls the error callback function when all logs fail", async function () {
+    const noOfElements = 25000;
+    const logData = getObjects(noOfElements);
+    const concurrency = 3;
+
+    let errorCallbackCount = 0;
+    const failedLogs: Record<string, unknown>[] = [];
+
+    function errorCallback(uploadLogsError: UploadLogsError): void {
+      if (
+        (uploadLogsError.cause as Error).message ===
+        "Data collection rule with immutable Id 'immutable-id-123' not found."
+      ) {
+        ++errorCallbackCount;
+        failedLogs.concat(uploadLogsError.failedLogs);
+      }
+    }
+
+    const uploadSinon = Sinon.spy(LogsIngestionClient.prototype, "upload");
+
+    try {
+      await client.upload("immutable-id-123", "Custom-MyTableRawData", logData, {
+        maxConcurrency: concurrency,
+        onError: errorCallback,
+      });
+    } catch (e: any) {
+      const result = isAggregateUploadLogsError(e) ? e.errors : [];
+      if (result.length > 0) {
+        result.forEach((err) => {
+          assert.equal(
+            err.cause.message,
+            `Data collection rule with immutable Id 'immutable-id-123' not found.`
+          );
+        });
+      }
+    }
+    assert.equal(errorCallbackCount, concurrency);
+    if (failedLogs.length > 0) {
+      try {
```

Comment:
why do we need try {} finally?

Reply:
In this test, where I am retrying the upload API, I don't want any failure to interrupt the flow and I want to ignore that, that's why I put a try finally block